### PR TITLE
Implement support for basic form data parameters

### PR DESCRIPTION
**Pull request by danielgtaylor**
This adds support for transforming `formData` parameters into data structure namespace elements attached to the request.

* [x] Support for basic types
* [x] Support for nested types (arrays with items)
* [x] Support for source maps for each generated element

cc @smizell

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 
+- Add support for `formData` parameters, which get converted into request data structures.
```
Comment (pksunkara) [inline Changelog.md]: There was a 0.5.0 release but it wasn't mentioned in the changelog.

**Comment by pksunkara** [comment]
There's an official W3 document on how JSON object keys are converted into uri parameters correctly. Please use that when adding support for complex types.

File: Changelog.md (hunk 1 above)
Comment (smizell) [inline Changelog.md]: I believe it's in master, and this will need to be rebased/changed to use those changes.

Comment (danielgtaylor) [inline Changelog.md]: I have rebased this locally, will push it up soon.

**Comment by danielgtaylor** [comment]
@smizell I believe this should be good to go now. The only open question is around content type - should I be setting the content type when I encounter `formData` parameters? Is there more than one allowable content type?

@pksunkara do you have a link? Is this something I need to do in the parser or is it done by other tooling?

**Comment by pksunkara** [comment]
@danielgtaylor I have failed to find the link. I remember it from W3 website and it was linked in some blueprint related github issue. Will try again tomorrow.

File: Changelog.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
-# 0.5.0 - 2015-12-8
+# Unreleased
 
+- Add support for `formData` parameters, which get converted into request data structures.
 - Support `x-summary` and `x-description` in Path Item Objects
```
Comment (smizell) [inline Changelog.md]: Is this already released?

File: src/adapter.js [hunk 3]
```diff
@@ -355,6 +396,8 @@ export function parse({minim, source, generateSourceMap}, done) {
     const setupSourceMap = makeSourceMap.bind(makeSourceMap, SourceMap, ast);
     const setupAnnotation = makeAnnotation.bind(makeAnnotation, Annotation,
       Link, SourceMap, ast, parseResult);
+    const paramToElement = convertParameterToMember.bind(
```
Comment (smizell) [inline src/adapter.js]: We are one past the rule of three now :)

Comment (smizell) [inline src/adapter.js]: Oh, take it back! I see this was copied from above.

**Comment by smizell** [comment]
This looks good to me :+1: 

The only thing is the changelog. Is it correct?

**Pull request merged**